questionnaires/templatetags/questionnaires_tags.py

```diff
@@ -68,7 +68,11 @@ def field_counter(form, forloop, form_length, fields_step, questionnaire):
 
 @register.inclusion_tag('questionnaires/tags/submit_button.html')
 def render_submit_button(fields_step, page):
-    return {"fields_step": fields_step, "page": page}
+    submit_button_text = page.submit_button_text
+    if hasattr(page, "multi_step") and page.multi_step and fields_step.paginator.num_pages != fields_step.number:
+        submit_button_text = "Next"
+
+    return {'submit_button_text': submit_button_text}
```

Comment:
This function is getting all of its information from the page, so why not move this to `Questionnaire`?

Is there a test for the logic in this function?

Reply:
This logic is used in surveys, quizzes, and embedded questionnaires. If we move it to the **questionnaire**, it will be repeated multiple times.

Reply:
I think we can work around that. What is required is a method to get the submit button text given a 'fields_step'. There can be a default implementation in QuestionnairePage (for Survey and Quiz), and an override in Poll.

There should be tests for this.